hdr_encoding.go

```diff
@@ -226,39 +226,125 @@ func (rh *Histogram) fillBufferFromCountsArray() (buffer []byte, err error) {
 
 // Read an LEB128 ZigZag encoded long value from the given buffer
 func zig_zag_decode_i64(buf []byte) (signedValue int64, n int) {
-	var value uint64 = 0
-	for shift := uint(0); ; shift += 7 {
-		if n >= len(buf) {
-			return 0, 0
-		}
-		b := uint64(buf[n])
-		n++
-		value |= (b & 0x7f) << shift
-		if (b & 0x80) == 0 {
-			break
+	if len(buf) < 1 {
+		return 0, 0
+	}
+	var value = uint64(buf[0]) & 0x7f
+	n = 1
+	if len(buf) > 1 && (buf[0]&0x80) != 0 {
```

Comment:
Error handling is probably needed here. When additional bytes are indicated (by buf[i]&0x80) but those bytes are not available in the buffer, returning a value without an error indication will lead to a silent reading of corrupt values. Truncated compressed histogram strings are a real thing (through various communication or buffer overflow situations), so these errors to happen and should be made visible to the caller.

Reply:
agree :+1: , will change the signature from `zig_zag_decode_i64(buf []byte) (signedValue int64, n int)` to `zig_zag_decode_i64(buf []byte) (signedValue int64, n int, err error)`

Reply:
fixed in the latest commits @giltene . Also added spefic test for this at https://github.com/HdrHistogram/hdrhistogram-go/pull/35/files#diff-39e1e370a64ae2eadefea4bdd36d4080R26